src/main/java/Duke.java

```diff
@@ -1,10 +1,585 @@
+import com.sun.source.util.TaskListener;
+
+import java.io.*;
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import static java.time.DayOfWeek.MONDAY;
+
 public class Duke {
+
+    public static String formatDate(LocalDate date) {
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
+        String formattedDate = date.format(outputFormatter);
+        return formattedDate;
+    }
+
+    public static String formatTime(LocalTime time) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("h:mm a");
+        String formattedTime = time.format(formatter);
+        return formattedTime;
+    }
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description, boolean isDone) {
+            this.description = description;
+            this.isDone = isDone;
+        }
+
+        public String getStatusIcon() {
+            return (this.isDone ? "X" : "0" ); // mark done task with 1
+        }
+
+        public String toString() {
+            return " | " + getStatusIcon() + " | " + this.description;
+        }
+
+        public void editDescription(String input) {
+            this.description = input;
+        }
+
+        public String checkType() {
+            return "Task";
+        }
+
+        public boolean checkDone() {
+            return this.isDone;
+        }
+        public void markDone() {
+            this.isDone = true;
+        }
+
+        public void markUnDone() {
+            this.isDone = false;
+        }
```

Comment:
You can merge these 2 functions into a single function, just by taking a boolean parameter